MultiSlider.js

```diff
@@ -1,573 +1,486 @@
-      });
-
-      if (value !== this.state.valueTwo) {
+      let value = positionToValue(confined, optionsArray, sliderLength);
+      let snapped = valueToPosition(value, optionsArray, sliderLength);
+
+      setPositionTwo(snapped ? snapped : confined);
+
+      if (value !== valueTwo) {
         this.setState(
           {
             valueTwo: value,
           },
           () => {
-            this.props.onValuesChange([
-              this.state.valueOne,
-              this.state.valueTwo,
-            ]);
-
-            this.props.onMarkersPosition([
-              this.state.positionOne,
-              this.state.positionTwo,
-            ]);
+            onValuesChange([valueOne, valueTwo]);
+
+            onMarkersPosition([positionOne, positionTwo]);
           },
         );
       }
     }
   };
 
-  endOne = gestureState => {
-    if (gestureState.moveX === 0 && this.props.onToggleOne) {
-      this.props.onToggleOne();
+  let endOne = gestureState => {
+    if (gestureState.moveX === 0 && onToggleOne) {
+      onToggleOne();
       return;
     }
 
     this.setState(
       {
-        pastOne: this.state.positionOne,
-        onePressed: !this.state.onePressed,
+        pastOne: positionOne,
+        onePressed: !onePressed,
       },
       () => {
-        var change = [this.state.valueOne];
-        if (this.state.valueTwo) {
-          change.push(this.state.valueTwo);
+        var change = [valueOne];
+        if (valueTwo) {
+          change.push(valueTwo);
         }
-        this.props.onValuesChangeFinish(change);
+        onValuesChangeFinish(change);
       },
     );
   };
 
-  endTwo = gestureState => {
-    if (gestureState.moveX === 0 && this.props.onToggleTwo) {
-      this.props.onToggleTwo();
+  let endTwo = gestureState => {
+    if (gestureState.moveX === 0 && onToggleTwo) {
+      onToggleTwo();
       return;
     }
 
     this.setState(
       {
-        twoPressed: !this.state.twoPressed,
-        pastTwo: this.state.positionTwo,
+        twoPressed: !twoPressed,
+        pastTwo: positionTwo,
       },
       () => {
-        this.props.onValuesChangeFinish([
-          this.state.valueOne,
-          this.state.valueTwo,
-        ]);
+        onValuesChangeFinish([valueOne, valueTwo]);
       },
     );
   };
 
-  componentDidUpdate(prevProps, prevState) {
-    const {
-      positionOne: prevPositionOne,
-      positionTwo: prevPositionTwo,
-    } = prevState;
+  let _panResponderBetween = customPanResponder(
+    gestureState => {
+      startOne(gestureState);
+      startTwo(gestureState);
+    },
+    gestureState => {
+      moveOne(gestureState);
+      moveTwo(gestureState);
+    },
+    gestureState => {
+      endOne(gestureState);
+      endTwo(gestureState);
+    },
+  );
+
+  let _panResponderOne = customPanResponder(startOne, moveOne, endOne);
 
-    const { positionOne, positionTwo } = this.state;
+  let _panResponderTwo = customPanResponder(startTwo, moveTwo, endTwo);
```

Comment:
useRef

Reply:
I want to make the component uncontrolled, but what do I have to remove/rewrite?